gtfslib/dao.py

```diff
@@ -73,12 +76,12 @@ def delete_feed(self, feed_id):
         self._session.query(FareRule).filter(FareRule.feed_id == feed_id).delete(synchronize_session=False)
         self._session.query(FareAttribute).filter(FareAttribute.feed_id == feed_id).delete(synchronize_session=False)
         self._session.query(ShapePoint).filter(ShapePoint.feed_id == feed_id).delete(synchronize_session=False)
-        self._session.query(Shape).filter(Shape.feed_id == feed_id).delete(synchronize_session=False)
         self._session.query(StopTime).filter(StopTime.feed_id == feed_id).delete(synchronize_session=False)
         self._session.query(Trip).filter(Trip.feed_id == feed_id).delete(synchronize_session=False)
+        self._session.query(Route).filter(Route.feed_id == feed_id).delete(synchronize_session=False)        
+        self._session.query(Shape).filter(Shape.feed_id == feed_id).delete(synchronize_session=False)                
         self._session.query(CalendarDate).filter(CalendarDate.feed_id == feed_id).delete(synchronize_session=False)
-        self._session.query(Calendar).filter(Calendar.feed_id == feed_id).delete(synchronize_session=False)
-        self._session.query(Route).filter(Route.feed_id == feed_id).delete(synchronize_session=False)
+        self._session.query(Calendar).filter(Calendar.feed_id == feed_id).delete(synchronize_session=False)        
```

Comment:
Thanks. I'm going to integrate this re-order change to solve the bug. I think we seldom see it because in SQLite there is no foreign-key integrity constraint, only in PostgreSQL.